gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/common/MongoFactory.java

```diff
@@ -310,6 +311,28 @@ private MongoClientSettings buildClientSettings(boolean isReactive) {
             .build();
     }
 
+    private MongoClientSettings buildClientSettingsFromURI(String uri, boolean isReactive) {
+        ServerSettings serverSettings = buildServerSettings();
+        SslSettings sslSettings = buildSslSettings();
+
+        MongoClientSettings.Builder settingsBuilder = MongoClientSettings
+            .builder()
+            .applyConnectionString(new ConnectionString(uri))
+            .applyToServerSettings(builder -> builder.applySettings(serverSettings))
+            .applyToSslSettings(builder -> builder.applySettings(sslSettings));
```

Comment:
Maybe we should call `applyConnectionString` at the end to prevent other settings to override the param in the URI ?

Reply:
True